File: Python/0025. Red And Green.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            brick = input().strip()
            l = len(brick)
            c = brick.count('R')
            res = c
            
            for s in brick:
                c += 1 if s=='G' else -1
                res = min(res,c)
            print(res)
        
        except EOFError:
            break
    
if __name__ == "__main__":
    func()

File: C++/0025. Red And Green.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
#include <algorithm>
int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 

    string brick;
    while(cin>>brick){
        int l = brick.size();
        int c = count(brick.begin(),brick.end(),'R');
        int res = c;
        for(auto s:brick){
            c += s=='G'?1:-1;
            res = min(res,c);
        }
        cout<<res<<endl;
    }
    return 0;
}
